crates/goose/src/logging.rs

```diff
@@ -0,0 +1,134 @@
+use anyhow::{Context, Result};
+use etcetera::{choose_app_strategy, AppStrategy};
+use std::fs;
+use std::path::PathBuf;
+
+use crate::config::APP_STRATEGY;
+
+/// Returns the base directory where log files should be stored.
+/// This is the common base for all goose logging.
+///
+/// Returns:
+/// - macOS/Linux: ~/.local/state/goose/logs
+/// - Windows:     ~\AppData\Roaming\Block\goose\data\logs
+pub fn get_base_log_directory() -> Result<PathBuf> {
+    let home_dir =
+        choose_app_strategy(APP_STRATEGY.clone()).context("HOME environment variable not set")?;
+
+    let base_log_dir = home_dir
+        .in_state_dir("logs")
+        .unwrap_or_else(|| home_dir.in_data_dir("logs"));
+
+    Ok(base_log_dir)
+}
+
+/// Returns the directory where log files should be stored for a specific component.
+/// Creates the directory structure if it doesn't exist.
+///
+/// # Arguments
+///
+/// * `component` - The component name (e.g., "cli", "server", "debug")
+/// * `use_date_subdir` - Whether to create a date-based subdirectory
+///
+/// # Returns
+///
+/// The path to the log directory for the specified component
+pub fn get_log_directory(component: &str, use_date_subdir: bool) -> Result<PathBuf> {
+    let base_log_dir = get_base_log_directory()?;
+    let component_dir = base_log_dir.join(component);
+
+    let log_dir = if use_date_subdir {
+        // Create date-based subdirectory
+        let now = chrono::Local::now();
+        component_dir.join(now.format("%Y-%m-%d").to_string())
+    } else {
+        component_dir
+    };
+
+    // Ensure log directory exists
+    fs::create_dir_all(&log_dir).context("Failed to create log directory")?;
+
+    Ok(log_dir)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::env;
+    use tempfile::TempDir;
+
+    fn setup_temp_home() -> TempDir {
+        let temp_dir = TempDir::new().unwrap();
+        if cfg!(windows) {
+            env::set_var("USERPROFILE", temp_dir.path());
+        } else {
+            env::set_var("HOME", temp_dir.path());
```

Comment:
I think tests that set env vars can cause issues, also this one doesn't seem to restore it to what it was before. At minimum probably needs a `#[serial]` if we must set them

Reply:
if we really want to unit test these, maybe have the main one read $HOME and pass that to another fn that we actually test. `get_base_log_directory` looks like it doesn't need to be `pub`, so you'd only need to do this in one place